### PR TITLE
Subscriptions: added command palette commands for quickly changing post access.

### PR DESCRIPTION
**Pull request by simison**
New command line commands to change posts into free, paid or subscribers-only. Commands show up only in the post editor, and not in page/site editors.

- When a post is made free, remove any paywall blocks from the content.

- When post is made paid, or subscribers-only, adds paywall block to the content after 1st block of the post.

- If the content is shorter than 2 blocks, add one empty block after the paywall block so that it's easier to continue writing paid content.

- If Stripe isn't connected, nudges to set up payments first.

- Has tracks events.

**Commands:**

Screenshot 2024-06-10 at 12 38 00

**Video showing how paywall block gets inserted/removed:**

https://github.com/Automattic/jetpack/assets/87168/055e88b6-a094-44bc-99cb-baf331b8bc98

**Notices:**

Screenshot 2024-06-07 at 15 51 59
Screenshot 2024-06-07 at 15 51 38

## Proposed changes:

*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* In post editor with subscriptions module enabled, press `cmd+k`
* Type "access" or some other keyword, and try different commands.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/command-palette.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			close();
+		},
+	} );
+
+	useCommand( {
+		name: 'jetpack/subscriptions-access-everyone',
+		label: __( 'Make post accessible to everyone', 'jetpack' ),
+		searchLabel: __( 'Make post accessible to everyone, free and paid subscribers', 'jetpack' ),
+		icon,
+		// context: 'site-editor', // @TODO: post editor only
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/command-palette.js]: TODO: limit commands only to post editor somehow.

**Comment by crisbusquets** [comment]
I tested it, and it works for me 🚢 

Regarding the icons (p1718018506585549/1717769305.344609-slack-C052XEUUBL4), I'd go with this one instead of the Jetpack icon.

It makes it clear what it does (inserts the Paywall block). Using the Jetpack icon for the three cases would also make it difficult to scan.

image

- Would it be possible that the icon is green instead of black? This way it would be more consistent with how the Jetpack-related icons show up when manually adding a block in Gutenberg
- Will we have metrics to see how many people use this?

**Comment by simison** [comment]
@crisbusquets agreed with icon choise reasoning. 👍 

I like the idea of applying the green, but the UI on the hover state suffers a bit:

image

Compared to how icon otherwise turns white on hover:
image

It might be enough to have Jetpack logo visible at Access panel, which is what this command essentially controls?

**Comment by simison** [comment]
Added tracks events.

**Comment by simison** [comment]
I think this is good to go — we can follow up with tweaking icon colours in separate PR.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/command-palette.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			const paywall = blocks[ 0 ];
+			if ( canRemoveBlock( paywall ) ) {
+				removeBlock( paywall );
+			}
+		}
+	}
+
+	function addPaywallBlock() {
+		// Check that paywall isn't already placed
+		const hasPaywallBlocks = getBlocksByName( paywallBlockName );
```
Comment (lezama) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/command-palette.js:66]: I wonder if `canInsertBlockType` already does this check, as the block is declared as `multiple = false`

Comment (simison) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/command-palette.js:66]: Good question!

But nope:

image

**Comment by simison** [comment]
Need to actually double-check the "if payments aren't set up yet, opens a payment setup modal" flow before merging, it was before some further changes when I tested that last time.

**Comment by simison** [comment]
Later iteration: hide or disable the "free" command when a post is already free, hide the "paid" command when a post is already paid, etc.

**Comment by enejb** [comment]
I am not sure if this a blocker but these commands also show up if you don't have newsletter features enabled. 
It might be good to only load the JS if that is the case. Or first check that the newsletter features are enabled.

**Comment by simison** [comment]
Good catch @enejb ! I added a check if the module is disabled. Ideally we wouldn't even load the code but that's a bit bigger change in how editor scripts in Jetpack work/load.

Screenshot 2024-06-12 at 10 13 32

**Review by lezama — APPROVED** [review]

**Pull request merged**